### DISCUSSION TITLE
How to specify metadata filename for downloaded file?

### DISCUSSION THREAD

**Discussion by aaricpittman** [shrinerb/shrine#683] [Q&A] [answered]
I have an ActiveRecord model that has an attachment named `file`.

```ruby
class Image < ApplicationRecord
  include ImageUploader::Attachment(:file)
end
```

What is the correct way to set that attachment when creating a new instance of the model using a file downloaded remotely and specifying the metadata filename?

Doing the following handles the file but because it is not a `UploadedFile` and the metadata filename is not set Shrine is unable to [determine the correct file extension](https://github.com/shrinerb/shrine/blob/bf43c6de28f4b6d58fa8c2d4ec22747db4e44e60/lib/shrine.rb#L275-L276).

```ruby
Image.create(file: Down.open("https://example.org/file"))
```

**Comment by adam12** [comment, marked as answer]
Untested, but you might need to go directly through the attacher for this.

```ruby
url = "https://example.org/file"
image = Image.new
image.file_attacher.attach(Down.open(url), metadata: { filename: File.basename(url) })
```

Alternatively the [`remote_url`](https://shrinerb.com/docs/plugins/remote_url) plugin?

**Reply by aaricpittman** [reply]
@adam12 `remote_url` is exactly what I needed. Thanks.
